### ISSUE TITLE
Ltsc2019 is also being chosen for Windows 10, HyperV not installed.

### ISSUE THREAD

**Issue by rdebath** [microsoft/navcontainerhelper#3417]
**Describe the issue**
Ltsc2019 is also being chosen for Windows 10, HyperV not installed.
Previously was using "2004".

**Scripts used to create container and cause the issue**
```
        New-Bcimage `
            -artifactUrl $artifactUrl `
            -imageName $navimage `
            -isolation process `
            -baseImage $baseImage `
            -includeTestToolkit:$tests -includeTestLibrariesOnly:$tests `
            -skipIfImageAlreadyExists

```
```
2024-03-14T07:25:04.1226600Z Generic Tag: 1.0.2.17
2024-03-14T07:25:04.1630572Z Container OS Version: 10.0.17763.5576 (ltsc2019)
2024-03-14T07:25:04.1645309Z Host OS Version: 10.0.19045.4170 (22H2)
2024-03-14T07:25:04.1701450Z WARNING: Host OS and Base Image Container OS doesn't match and process isolation is specified. If you encounter issues, you could try to specify -isolation hyperv
2024-03-14T07:25:04.1709749Z Using process isolation
```
```
2024-03-14T07:26:37.1143303Z Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly
2024-03-14T07:26:37.3328832Z  ---> Running in 82225614aff6
2024-03-14T07:26:40.4013128Z DockerDo : hcs::CreateComputeSystem 82225614aff6b80974637bd17d8556a3a9486501ceba3d3735105199e0600956: The container 
2024-03-14T07:26:40.4014374Z operating system does not match the host operating system.
2024-03-14T07:26:40.4014716Z ExitCode: 1
2024-03-14T07:26:40.4016360Z Commandline: docker build --isolation=process --memory 8G --no-cache --tag navimage:daily-gb.4 
2024-03-14T07:26:40.4017087Z c:\bcartifacts.cache\yzxzhwh5.yym
2024-03-14T07:26:40.4018958Z At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.9\ContainerHandling\New-NavImage.ps1:715 char:27
2024-03-14T07:26:40.4020790Z + ...       if (!(DockerDo -command build -parameters @("--isolation=$isola ...
2024-03-14T07:26:40.4021305Z +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-03-14T07:26:40.4021925Z     + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
2024-03-14T07:26:40.4022526Z     + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo
2024-03-14T07:26:40.4023071Z  
2024-03-14T07:26:40.4984930Z ##[error]PowerShell exited with code '1'.
```

**Full output of scripts**
```
2024-03-14T07:24:39.7258808Z ##[section]Starting: Create Build Image, Daily
2024-03-14T07:24:39.7352967Z ==============================================================================
2024-03-14T07:24:39.7353201Z Task         : PowerShell
2024-03-14T07:24:39.7353297Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2024-03-14T07:24:39.7353538Z Version      : 2.236.2
2024-03-14T07:24:39.7353618Z Author       : Microsoft Corporation
2024-03-14T07:24:39.7353749Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2024-03-14T07:24:39.7354080Z ==============================================================================
2024-03-14T07:24:40.5443470Z Generating script.
2024-03-14T07:24:40.5945400Z Formatted command: . 'C:\Agents\Agent1\_work\8\s\Self\.azureDevOps\Scripts\Create-Image.ps1' -bcSelect Daily
2024-03-14T07:24:40.6224605Z ========================== Starting Command Output ===========================
2024-03-14T07:24:40.6444624Z ##[command]"C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\Agents\Agent1\_work\_temp\d87ce9bd-afc9-4031-86f7-6a7c1d905902.ps1'"
2024-03-14T07:24:40.8638939Z ### Artifact country gb ###
2024-03-14T07:24:40.8652398Z ### Artifact version  ###
2024-03-14T07:24:40.8660200Z ### Artifact type sandbox ###
2024-03-14T07:24:40.8667937Z ### Artifact select Daily ###
2024-03-14T07:24:40.8762942Z ### Using Docker image navimage:daily-gb.4 ###
2024-03-14T07:24:40.9175760Z >>> Locking image build  + Global\RunningBCImage
2024-03-14T07:24:41.0463191Z BcContainerHelper version 6.0.9
2024-03-14T07:24:41.3525491Z Setting MicrosoftTelemetryConnectionString = 
2024-03-14T07:24:41.4719358Z Running on Windows, PowerShell 5.1.19041.4170
2024-03-14T07:25:03.2402929Z ### Using Artifact  https://bcartifacts.azureedge.net/sandbox/23.5.16502.17362/gb  ###
2024-03-14T07:25:03.5047623Z Untagged: navimagev:22.5.59966.63884-gb
2024-03-14T07:25:03.5088857Z Untagged: navimagev:24.0.16410.17172-gb
2024-03-14T07:25:03.5131935Z ### Running New-Bcimage ###
2024-03-14T07:25:03.6499872Z ### Using base image  mcr.microsoft.com/businesscentral:ltsc2019  ###
2024-03-14T07:25:03.8195935Z navimage:daily-gb.4
2024-03-14T07:25:03.8210172Z Fetching all docker images
2024-03-14T07:25:03.8639140Z Building image navimage:daily-gb.4 based on mcr.microsoft.com/businesscentral:ltsc2019 with https://bcartifacts.azureedge.net/sandbox/23.5.16502.17362/gb
2024-03-14T07:25:03.8646170Z Pulling latest image mcr.microsoft.com/businesscentral:ltsc2019
2024-03-14T07:25:04.0596773Z ltsc2019: Pulling from businesscentral
2024-03-14T07:25:04.0609375Z Digest: sha256:ca49064a275b9e95c8d3a0ae8264a4222cd7597a84dd6d007276151cfe6bebfa
2024-03-14T07:25:04.0617823Z Status: Image is up to date for mcr.microsoft.com/businesscentral:ltsc2019
2024-03-14T07:25:04.0642619Z mcr.microsoft.com/businesscentral:ltsc2019
2024-03-14T07:25:04.1226600Z Generic Tag: 1.0.2.17
2024-03-14T07:25:04.1630572Z Container OS Version: 10.0.17763.5576 (ltsc2019)
2024-03-14T07:25:04.1645309Z Host OS Version: 10.0.19045.4170 (22H2)
2024-03-14T07:25:04.1701450Z WARNING: Host OS and Base Image Container OS doesn't match and process isolation is specified. If you encounter issues, you could try to specify -isolation hyperv
2024-03-14T07:25:04.1709749Z Using process isolation
2024-03-14T07:25:04.1816667Z Files in c:\bcartifacts.cache\yzxzhwh5.yym\my:
2024-03-14T07:25:04.1884640Z Copying Platform Artifacts
2024-03-14T07:25:04.1945727Z c:\bcartifacts.cache\sandbox\23.5.16502.17362\platform
2024-03-14T07:25:09.8804239Z Copying Database
2024-03-14T07:25:12.0200559Z Copying Licensefile
2024-03-14T07:25:12.0338653Z Copying ConfigurationPackages
2024-03-14T07:25:12.0347186Z C:\bcartifacts.cache\sandbox\23.5.16502.17362\gb\ConfigurationPackages
2024-03-14T07:25:12.1108935Z Copying Extensions
2024-03-14T07:25:12.1120664Z C:\bcartifacts.cache\sandbox\23.5.16502.17362\gb\Extensions
2024-03-14T07:25:13.0722937Z Copying Applications.GB
2024-03-14T07:25:13.0731652Z C:\bcartifacts.cache\sandbox\23.5.16502.17362\gb\Applications.GB
2024-03-14T07:25:13.5219223Z c:\bcartifacts.cache\yzxzhwh5.yym
2024-03-14T07:25:31.6093510Z Sending build context to Docker daemon  2.567GB
2024-03-14T07:25:31.6102225Z 
2024-03-14T07:25:31.6887038Z Step 1/6 : FROM mcr.microsoft.com/businesscentral:ltsc2019
2024-03-14T07:25:31.6894396Z  ---> 64568c89e98d
2024-03-14T07:25:31.6902752Z Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/23.5.16502.17362/gb filesOnly=False
2024-03-14T07:25:31.9176364Z  ---> Running in 9fbf1c195d93
2024-03-14T07:25:33.1665717Z  ---> Removed intermediate container 9fbf1c195d93
2024-03-14T07:25:33.1680590Z  ---> 191973ccf1ac
2024-03-14T07:25:33.1692746Z Step 3/6 : COPY my /run/
2024-03-14T07:25:34.4380398Z  ---> d7e320034195
2024-03-14T07:25:34.4391918Z Step 4/6 : COPY NAVDVD /NAVDVD/
2024-03-14T07:26:37.1130444Z  ---> 03bbf76a6031
2024-03-14T07:26:37.1143303Z Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly
2024-03-14T07:26:37.3328832Z  ---> Running in 82225614aff6
2024-03-14T07:26:40.4013128Z DockerDo : hcs::CreateComputeSystem 82225614aff6b80974637bd17d8556a3a9486501ceba3d3735105199e0600956: The container 
2024-03-14T07:26:40.4014374Z operating system does not match the host operating system.
2024-03-14T07:26:40.4014716Z ExitCode: 1
2024-03-14T07:26:40.4016360Z Commandline: docker build --isolation=process --memory 8G --no-cache --tag navimage:daily-gb.4 
2024-03-14T07:26:40.4017087Z c:\bcartifacts.cache\yzxzhwh5.yym
2024-03-14T07:26:40.4018958Z At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.9\ContainerHandling\New-NavImage.ps1:715 char:27
2024-03-14T07:26:40.4020790Z + ...       if (!(DockerDo -command build -parameters @("--isolation=$isola ...
2024-03-14T07:26:40.4021305Z +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-03-14T07:26:40.4021925Z     + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
2024-03-14T07:26:40.4022526Z     + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo
2024-03-14T07:26:40.4023071Z  
2024-03-14T07:26:40.4984930Z ##[error]PowerShell exited with code '1'.
2024-03-14T07:26:40.5259541Z ##[section]Finishing: Create Build Image, Daily```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
- does it happen all the time?
- did it use to work?

**Comment by freddydk** [comment]
That is correct.
In order to use Microsoft official images going forward, you will have to use HyperV on Windows 10.
Reason for this is, that I can only build new images for supported OS' - Microsoft policy.

You can however build the generic image for 2004 yourself, by cloning https://github.com/microsoft/nav-docker and running the build.ps1 script in the generic folder.

This will build **mygeneric:latest** on your machine for the version you specify - and you would have to specify that in your settings file or in the useGenericImageName parameter.

You can also push this to the docker hub to your account and use it from there.
Note that when there are updates to nav-docker - it should be re-built.

Not ideal, but possible.

**Comment by rdebath** [comment]
Okay, I've found that my actual issue is that [the Microsoft docker install script](https://github.com/microsoft/Windows-Containers/blob/Main/helpful_tools/Install-DockerCE/install-docker-ce.ps1) doesn't actually turn on Hyper-V despite claiming that it is already turned on. Once it's turned on the provided docker successfully uses Hyper-V to create a real VM to run the image.

So I've discarded the remaining Win10 VM agents and am installing Hyper-V on the old laptops.

`Enable-WindowsOptionalFeature -Online -FeatureName "Microsoft-Hyper-V" -All`

BTW: Yes a `mygeneric` image works fine.
